akka-docs/src/main/paradox/typed/persistence-durable-state.md

```diff
@@ -0,0 +1,391 @@
+
+Java
+:  @@snip [DurableStatePersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java) { #behavior }
+
+## Effects and Side Effects
+
+A command handler returns an `Effect` directive that defines what state, if any, to persist. 
+Effects are created using @java[a factory that is returned via the `Effect()` method] @scala[the `Effect` factory]
+and can be one of: 
+
+* `persist` will persist state atomically, i.e. all states are stored or none of them are stored if there is an error
+* `none` no state to be persisted, for example a read-only command
+* `unhandled` the command is unhandled (not supported) in current state
+* `stop` stop this actor
+* `stash` the current command is stashed
+* `unstashAll` process the commands that were stashed with @scala[`Effect.stash`]@java[`Effect().stash`]
+* `reply` send a reply message to the given `ActorRef`
+
+Note that only one of those can be chosen per incoming command. It is not possible to both persist and say none/unhandled.
+
+In addition to returning the primary `Effect` for the command `DurableStateBehavior`s can also 
+chain side effects that are to be performed after successful persist which is achieved with the `thenRun`
+function that runs the callback passed to it e.g @scala[`Effect.persist(..).thenRun`]@java[`Effect().persist(..).thenRun`]. 
+
+All `thenRun` registered callbacks are executed sequentially after successful execution of the persist statement
+(or immediately, in case of `none` and `unhandled`).
+
+In addition to `thenRun` the following actions can also be performed after successful persist:
+
+* `thenStop` the actor will be stopped
+* `thenUnstashAll` process the commands that were stashed with @scala[`Effect.stash`]@java[`Effect().stash`]
+* `thenReply` send a reply message to the given `ActorRef`
+
+In the example below, we use a different constructor of `DurableStateBehavior.withEnforcedReplies`, which creates
+a `Behavior` for a persistent actor that ensures that replies to commands are not forgotten. Hence it will be
+a compilation error if the returned effect isn't a `ReplyEffect`.
+
+Instead of `Increment` we will have a new command `IncrementWithConfirmation` that, along with persistence will also
+send an acknowledgement as a reply to the `ActorRef` passed in the command. 
+
+Example of effects and side-effects:
+
+Scala
+:  @@snip [DurableStatePersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/DurableStatePersistentBehaviorCompileOnly.scala) { #effects }
+
+Java
+:  @@snip [DurableStatePersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java) { #effects }
+
+Most of the time this will be done with the `thenRun` method on the `Effect` above. You can factor out
```

Comment:
Not very clear what this refers to, better to repeat something like 'It is useful to define side effects with `thenRun` ...'

Reply:
rephrased